packages/web-components/src/components/leadspace/leadspace.ts

```diff
@@ -0,0 +1,223 @@
+    return this.defaultSrc && this.type === LEADSPACE_TYPE.CENTERED ? `background-image: url(${this.defaultSrc})` : '';
+  }
+
+  /**
+   * Returns a class-name based on the gradient parameter type
+   */
+  protected _getGradientClass() {
+    return this.gradient ? `${prefix}--leadspace--gradient` : '';
+  }
+
+  /**
+   * Returns a class-name based on the type parameter type
+   */
+  protected _getTypeClass() {
+    let classname = '';
+
+    if (this.type === 'centered') {
+      classname += `${prefix}--leadspace--centered `;
+
+      if (this._hasImage) {
+        classname += `${prefix}--leadspace--centered__image `;
+      }
+    }
+
+    if (this.theme) {
+      classname += `${prefix}--leadspace--${this.theme} `;
+    }
+
+    if (this.type === 'small') {
+      classname += `${prefix}--leadspace--productive `;
+    }
+    return classname;
+  }
+
+  /**
+   * Renders Leadspace copy/description
+   */
+  protected _renderCopy() {
+    const { copy } = this;
+    return html`
+      <div class="${prefix}--leadspace__row">
+        <p data-autoid="${ddsPrefix}--leadspace__desc" class="${prefix}--leadspace__desc">
+          <slot name="copy">${copy}</slot>
+        </p>
+      </div>
+    `;
+  }
+
+  /**
+   * Renders Leadspace title slot
+   */
+  protected _renderTitle() {
+    const { title } = this;
+    return html`
+      <slot name="title">${title}</slot>
+    `;
+  }
+
+  /**
+   *  Renders the image slot or the mobile image for centered leadspace
+   */
+  protected _renderImage() {
+    const { defaultSrc, alt, type } = this;
+    if (type === LEADSPACE_TYPE.CENTERED) {
+      return html`
+        <div data-autoid="${ddsPrefix}--leadspace--centered--mobile__image" class="${prefix}--leadspace--centered--mobile__image">
+          <img src="${defaultSrc}" alt="${alt}" />
+        </div>
+      `;
+    }
+    return html`
+      <slot name="image" @slotchange="${this._handleSlotChangeImage}"></slot>
+    `;
+  }
+
+  /**
+   * Handles `slotchange` event on `<slot name="image">`.
+   *
+   * @param event The event.
+   */
+  private _handleSlotChangeImage(event: Event) {
+    this._hasImage = (event.target as HTMLSlotElement)
+      .assignedNodes()
+      .some(node => node.nodeType !== Node.TEXT_NODE || node!.textContent!.trim());
+  }
+
+  /**
+   * `true` if there is image content.
+   */
+  @internalProperty()
+  private _hasImage = false;
+
+  /**
+   * The alternate text.
+   */
+  @property()
+  alt = '';
+
+  /**
+   * The image source.
+   */
+  @property({ attribute: 'default-src' })
+  defaultSrc = '';
+
+  /**
+   * The leadspace copy.
+   */
+  @property()
+  copy = '';
+
+  /**
+   * The leadspace title.
+   */
+  @property()
+  title = '';
+
+  /**
+   *  Leadspace type (small, left, or centered)
+   */
+  @property({ reflect: true })
+  type = LEADSPACE_TYPE.LEFT;
+
+  /**
+   * Carbon color theme of the Leadspace
+   */
+  @property({ reflect: true })
+  theme = LEADSPACE_THEME;
```

Comment:
Continuing on [the discussion on `theme` and CSS custom properties](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/4084#discussion_r497258273); Wondering if theme chooser introduced some weeks ago (#4106) is enough for showcase the leadspace with a particular theme, or `theme` attribute/property is still needed.

Reply:
hm yeah I think you're right, I'll remove the `theme` property! Thanks!

Reply:
Splendid - Thanks @annawen1!

Reply:
I ran into issues with the gradient styles, specifically with the `rgba()` method where it doesn't grab the value for `$ui-background` token in time so I had to hardcode the colors. So Im thinking of keeping the theming property for now

Reply:
@annawen1 Thank you for taking a look. Would you able to use [`getComputedStyle()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) to query for `--ui-background` so that we can turn `theme` to an internal (private) property? If it sounds too much please let me know and I can take some look.

Reply:
I see, I'm not too sure how to implement that approach, should we merge this PR first and then take a look on how to do that?

Reply:
Makes sense @annawen1! Let's go this way.